### PR TITLE
Add Tabular Islamic Calendar

### PR DESCRIPTION
**Pull request by sotam1069**

### CODE REVIEW

File: components/calendar/src/islamic.rs [hunk 1]
```diff
@@ -824,6 +824,250 @@ impl DateTime<IslamicCivil> {
...
+    /// Constructs a new Islamic Tabular Calendar
+    pub fn new() -> Self {
+        Self
+    }
+
+    // "Fixed" is a day count representation of calendars staring from Jan 1st of year 1 of the Georgian Calendar.
+    // The fixed date algorithms are from
+    // Dershowitz, Nachum, and Edward M. Reingold. _Calendrical calculations_. Cambridge University Press, 2008.
+    //
+    // Lisp code reference:https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2076
```
Comment (echeran) [inline components/calendar/src/islamic.rs:967]:
```suggestion
    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2076
```

File: components/calendar/src/islamic.rs [hunk 2]
```diff
@@ -1382,13 +1794,21 @@ mod test {
...
     #[test]
-    fn test_islamic_epoch() {
+    fn test_islamic_epoch_friday() {
         let epoch = FIXED_ISLAMIC_EPOCH_FRIDAY.to_i64_date();
         // Iso year of Islamic Epoch
         let epoch_year_from_fixed = Iso::iso_from_fixed(RataDie::new(epoch)).inner.0.year;
         // 622 is the correct ISO year for the Islamic Epoch
         assert_eq!(epoch_year_from_fixed, 622);
     }
+    #[test]
```
Comment (echeran) [inline components/calendar/src/islamic.rs:1804]: newline

**Review by echeran — COMMENTED** [review]
LGTM modulo nitpicks

File: components/calendar/src/islamic.rs [hunk 3]
```diff
@@ -824,6 +824,250 @@ impl DateTime<IslamicCivil> {
...
 }
 
+/// The inner date type used for representing [`Date`]s of [`IslamicTabular`]. See [`Date`] and [`IslamicTabular`] for more details.
+
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
+pub struct IslamicTabularDateInner(ArithmeticDate<IslamicTabular>);
+
+impl CalendarArithmetic for IslamicTabular {
+    fn month_days(year: i32, month: u8) -> u8 {
+        match month {
```
Comment (Manishearth) [inline components/calendar/src/islamic.rs:834]: wow so nice and simple :smile:

**Review by Manishearth — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**